docs/extend/packaging-zos-extensions.md

```diff
@@ -0,0 +1,95 @@
+  * **`branch`**: It tells the user which branch this package is built from.
+  * **`number`**: You may create multiple packages in same branch. This is the sequential number of the current package.
+  * **`commitHash`**: This is the commit hash of the package we can match the exact source code in repository. Zowe core components usually uses `git rev-parse --verify HEAD` to retrieve commit hash.
+  * **`timestamp`**: This is the unix timestamp when the package is created.
+- **`commands`**: This field is high level of several sub-fields to define actions should be taken when the component is installed, configured, started or tested. All sub-fields are optional and usually should be pointed to a USS command or script.
+  * **`install`**: This defines extra steps when installing this component. It will be automatically executed if you install your component with `<RUNTIME_DIR>/bin/zowe-install-component.sh` utility tool.
+  * **`configureInstance`**: This defines extra steps when configure the component for a Zowe instance. It will be automatically executed if you configure your component with `<RUNTIME_DIR>/bin/zowe-configure-component.sh` utility tool.
+  * **`validate`**: This defines extra validations the component requires other than global validations. It is for runtime purpose, and will be automatically executed every time when Zowe is started.
+  * **`configure`**: This defines extra configuration steps before starting the component. It is for runtime purpose, and will be automatically executed every time  when Zowe is started.
+  * **`start`**: This tells Zowe launch script how to start the component. It is for runtime purpose, and will be automatically executed every time  when Zowe is started.
+- **`apimlServices`**: This section defines how the component will be registered to API Mediation Layer Discovery Service. All sub-fields are optional.
+  * **`dynamic`**: Array of objects. This information will tell Zowe and users what services you will register under Discovery service.
+    - **`serviceId`**: This defines the service ID registered to Discovery Service.
+  * **`static`**: Array of objects. If the component is statically registered under Discovery service, this tells Zowe where to find these static definitions. This information is for Zowe runtime. When Zowe is started, the launch script will check this field and put the parse static definition file into the directory defined as `STATIC_DEF_CONFIG_DIR` in Zowe instance.
+    - **`file`**: Path to the static definition file. This file is supposed to be a template.
+- **`desktopPlugins`**: Array of objects. This section defines how the component will be registered to Desktop Plugin. All sub-fields are optional.
```

Comment:
i think its only `desktopPlugin` singular rather than plural

https://github.com/JadinLuong/zowe-install-packaging/blob/ad20c9d8aff8ce0d610d1fc9b95c380995d5f932/bin/zowe-configure-component.sh#L85

Reply:
Thanks Nakul. Good catch, `desktopPlugins` is the intention. I just fixed the other PR.